### PR TITLE
Add essl to component manager

### PR DESCRIPTION
**Pull request by ginkgm**
Moving this component from IDF to extra components, since it's upper layer component.

Tracking ESP-IDF commit: [4f1a9e436e](https://github.com/espressif/esp-idf/commit/4f1a9e436e)

### CODE REVIEW

**Comment by ginkgm** [comment]
@mahavirj  @tore-espressif  PTAL

File: esp_serial_slave_link/README.md
Comment (mahavirj) [inline esp_serial_slave_link/README.md:9]: Do we plan to maintain documentation of this component on IDF side? I think this might be little inconsistent as any change here can break the documentation. Can we consider moving documentation to component here?

File: esp_serial_slave_link/essl.c
Comment (mahavirj) [inline esp_serial_slave_link/essl.c:1]: Some files here have copyright in SPDX format while some older one. Suggest that we use same format for all files in this component.

File: esp_serial_slave_link/idf_component.yml
Comment (mahavirj) [inline esp_serial_slave_link/idf_component.yml:1]: If API here is finalized then we can start with version "1.0.0"

File: esp_serial_slave_link/idf_component.yml
Comment (mahavirj) [inline esp_serial_slave_link/idf_component.yml:5]: I think component manager related workflow was not enabled for v4.3 case (e.g. `idf.py add-dependency`). Could you please check and then change this to `idf: ">=4.4"`?

File: esp_serial_slave_link/README.md
Comment (mahavirj) [inline esp_serial_slave_link/README.md:7]:
```suggestion
The port layer (`essl_sdio.c/essl_spi.c`) are currently only written to run on ESP chips in master mode, but you may also modify them to work on more platforms.
```

**Review by mahavirj — CHANGES_REQUESTED** [review]
@ginkgm Please add entry for this component in `.github/workflows/upload_component.yml`

File: esp_serial_slave_link/README.md
Comment (ginkgm) [reply esp_serial_slave_link/README.md:9]: Not sure which one is better actually. I would suggest to move the part of documentation here after this repo has some documentation system

File: esp_serial_slave_link/idf_component.yml
Comment (ginkgm) [reply esp_serial_slave_link/idf_component.yml:5]: OK.. I just copied it from some other component..

File: esp_serial_slave_link/README.md
Comment (mahavirj) [reply esp_serial_slave_link/README.md:9]: @igrr What would you recommend here?

File: esp_serial_slave_link/idf_component.yml
Comment (tore-espressif) [inline esp_serial_slave_link/idf_component.yml:2]:
```suggestion
description: "Espressif Serial Slave Link Library"
```

**Review by tore-espressif — COMMENTED** [review]
Would it make sense to move tests here too?
e.g. https://github.com/espressif/esp-idf/blob/master/components/driver/test/test_spi_slave_hd.c
https://github.com/espressif/esp-idf/blob/master/components/driver/test/test_sdio.c

We are working on running unit tests in GitHub action, ~~but it is not ready yet. As of now, we only build the tests~~

**EDIT:** See #13 . This PR runs test_app on our self-hosted runners

**Comment by tore-espressif** [comment]
Please rebase on top of https://github.com/espressif/idf-extra-components/pull/12 , it checks formating in GH action.

And let me know whether you are ready to merge

**Comment by ginkgm** [comment]
I think the target-test is too complicated to be in this repo. So I will only add a simple build test, to make sure the component can be build fine.

**Review by suda-morris — APPROVED** [review]

**Comment by igrr** [comment]
@ginkgm could you please check the two static analysis warnings reported above?

File: esp_serial_slave_link/include/esp_serial_slave_link/essl_sdio.h
Comment (tore-espressif) [inline esp_serial_slave_link/include/esp_serial_slave_link/essl_sdio.h:22]: Missing C++ guard macro

File: esp_serial_slave_link/include/esp_serial_slave_link/essl.h
Comment (tore-espressif) [inline esp_serial_slave_link/include/esp_serial_slave_link/essl.h:14]: Missing C++ guard macro

File: esp_serial_slave_link/include/esp_serial_slave_link/essl.h
Comment (tore-espressif) [inline esp_serial_slave_link/include/esp_serial_slave_link/essl.h:12]: These includes are not used in this file

File: esp_serial_slave_link/include/esp_serial_slave_link/essl_sdio.h
Comment (tore-espressif) [inline esp_serial_slave_link/include/esp_serial_slave_link/essl_sdio.h:14]: Freertos includes unused

File: esp_serial_slave_link/essl_sdio.c
Comment (tore-espressif) [inline esp_serial_slave_link/essl_sdio.c:480]: You can solve your build errors by using macros from `inttypes.h`
```suggestion
    ESP_LOGV(TAG, "send_slave_intr: %02" PRIx32, intr_mask);
```

**Review by tore-espressif — CHANGES_REQUESTED** [review]
@ginkgm Thank you for following this up.

I'm haven't worked with essl before, so I can't review properly.
Just make sure that the code compiles with esp-idf master

File: esp_serial_slave_link/idf_component.yml
Comment (ginkgm) [reply esp_serial_slave_link/idf_component.yml]: This seems not able to build with v4.4 for now. I set the version to v5.0. Then I may extend the support to v4.4 if necessary.

**Comment by ginkgm** [comment]
@tore-espressif @kumekay @mahavirj Please merge this MR if it's OK for you. There is basic build test for the source files.

For the compatibilities, I tried with ESP-IDF master examples with slight changes. Those changes will be added to IDF master after this is merged.

**Review by mahavirj — APPROVED** [review]

**Pull request merged**
